File: javadevelopers-understanding-kotlin-properties/src/test/java/com/moelholm/jav/OneTests.java
package com.moelholm.jav;

import static org.assertj.core.api.Assertions.assertThat;

import com.moelholm.One;
import org.junit.Test;

public class OneTests {

  @Test
  public void test() {

    One one = new One("Hello from Java");

    assertThat(one.getMessage()).isEqualTo("Hello from Java");

    // This won't compile, as there is no setter:

    // one.setMessage("Hello from Java");

  }

}

File: javadevelopers-understanding-kotlin-properties/src/test/kotlin/com/moelholm/kot/OneTests.kt
package com.moelholm.kot

import com.moelholm.One
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class OneTests {

  @Test
  fun test() {

    val one = One("Hello from Kotlin")

    assertThat(one.message).isEqualTo("Hello from Kotlin")

    // This won't compile, as there is no setter:

    // one.message = "Hello from Kotlin";

  }

}